feathr-impl/src/main/scala/com/linkedin/feathr/offline/transformation/DataFrameExt.scala

```diff
@@ -38,5 +39,54 @@ private[offline] object DataFrameExt {
       val extendedOtherDf = extendDf(other, schema)
       extendedDf.unionByName(extendedOtherDf)
     }
+
+
+    /**
+     * Copy the value of rightJoinColumns from rightDF to the current DataFrame, and rename as leftJoinColumns columns
+     * e.g.
+     * left          right
+     * id1 id2 v1 v2       id3 id4 v3
+     * 0    5  a   b        0   5   e
+     * 1    6  c   d        10  11  f
+     *
+     * will return:
+     * id1 id2 v1 v2
+     * 0  5  a  b
+     * 1  6  c  d
+     * 0  5  a  b
+     * 10  11  c d
+     *
+     * @param leftJoinColumns
+     * @param rightJoinColumns
+     * @param rightDF
+     * @return
+     */
+    def appendRows(leftJoinColumns: Seq[String], rightJoinColumns: Seq[String], rightDF: DataFrame): DataFrame = {
+      // Add a few more rows to observation data by copying the feature key from the feature dataset
+      // This ensures the join will have matched results between observation and feature data
+      val dfCount = df.count()
```

Comment:
Would'nt this be very expensive?

Reply:
It should be fine as we are in sanity check mode where the dataframe typically has been down sampled to a few rows.